packages/obojobo-document-engine/server/attempt-end.js

```diff
@@ -334,119 +337,127 @@ const insertAttemptScoredEvents = (
+// 	const temp = []
+// 	for (const node in state.chosen) {
+// 		if (node.type === QUESTION_NODE_TYPE) {
+// 			temp.push(assessmentProperties.oboNode.draftTree.getChildNodeById(node.id).toObject())
+// 		}
+// 	}
+// 	state.chosen = temp
+
+// 	// If reviews are always allowed, reload the state for this attempt
+// 	// Each attempt's state will be reloaded as it finishes
+// 	if (assessmentNode.node.content.review === 'always') {
+// 		return Assessment.updateAttemptState(attemptId, state)
+// 	}
+
+// 	// If reviews are allowed after last attempt and this is the last attempt,
+// 	// reload the states for all attempts
+// 	if (assessmentNode.node.content.review === 'no-attempts-remaining' && isLastAttempt) {
+// 		// Reload state for all previous attempts
+// 		return Assessment.getAttempts(
+// 			assessmentProperties.user.id,
+// 			draftId,
+// 			isPreview,
+// 			assessmentProperties.id
+// 		).then(result => {
+// 			result.attempts.map(attempt => {
+// 				attempt.state.qb = recreateChosenQuestionTree(
+// 					attempt.state.qb,
+// 					assessmentProperties.draftTree
+// 				)
+
+// 				const newQuestions = []
+
+// 				attempt.state.chosen.forEach(question => {
+// 					newQuestions.push(getNodeQuestion(question.id, assessmentProperties.draftTree))
+// 				})
+
+// 				attempt.state.chosen = newQuestions
+
+// 				return Assessment.updateAttemptState(attempt.attemptId, attempt.state)
+// 			})
+// 		})
+// 	}
+
+// 	logger.error(`Error: Reached exceptional state while reloading state for ${attemptId}`)
+// 	return null
+// }
+
+// const recreateChosenQuestionTree = (node, assessmentNode) => {
+// 	if (node.type === QUESTION_NODE_TYPE) {
+// 		return getNodeQuestion(node.id, assessmentNode)
+// 	}
+
+// 	const newChildren = []
+
+// 	for (const child of node.children) {
+// 		newChildren.push(recreateChosenQuestionTree(child, assessmentNode))
+// 	}
+
+// 	node.children = newChildren
+// 	return node
+// }
 
-	for (const child of node.children) {
-		newChildren.push(recreateChosenQuestionTree(child, assessmentNode))
-	}
-
-	node.children = newChildren
-	return node
-}
 // Pulls down a single question from the draft
 const getNodeQuestion = (nodeId, assessmentNode) => {
 	return assessmentNode.getChildNodeById(nodeId).toObject()
 }
 
 // Pulls assessment properties out of the promise flow
-const loadAssessmentProperties = (draftTree, attempt, user, isPreview, attemptHistory) => {
-	const assessmentNode = draftTree.getChildNodeById(attempt.assessmentId)
-
-	return {
-		user: user,
-		isPreview: isPreview,
-		draftTree: draftTree,
-		id: attempt.assessmentId,
-		oboNode: assessmentNode,
-		nodeChildrenIds: assessmentNode.children[1].childrenSet,
-		questionBank: assessmentNode.children[1],
-		assessmentQBTree: assessmentNode.children[1].toObject(),
-		attemptHistory: attemptHistory,
-		numAttemptsTaken: null,
-		childrenMap: null
-	}
-}
+// const loadAssessmentProperties = (draftTree, attempt, user, isPreview, attemptHistory) => {
```

Comment:
This is no longer used, but I want to keep it in the PR in case I unknowingly deleted important functionality. Will remove be pull.